src/Featurizers/Structs.h

```diff
@@ -1,245 +1,329 @@
+    std::uint64_t const                     NumElements;
+    std::vector<ValueEncoding> const        Values;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    SparseVectorEncoding(std::uint64_t numElements, std::vector<ValueEncoding> values);
+
+    bool operator==(SparseVectorEncoding const &other) const;
+    bool operator!=(SparseVectorEncoding const &other) const;
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         SingleValueSparseVectorEncoding
+///  \brief         Object that contains information that can be used to create
+///                 a sparse vector with only a single value set without creating
+///                 the vector itself.
+///
+template <typename T>
+class SingleValueSparseVectorEncoding {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using value_type                        = T;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    std::uint64_t const                     NumElements;
+    value_type const                        Value;
+    std::uint64_t const                     Index;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    SingleValueSparseVectorEncoding(std::uint64_t numElements, value_type value, std::uint64_t index);
+
+    bool operator==(SingleValueSparseVectorEncoding const &other) const;
+    bool operator!=(SingleValueSparseVectorEncoding const &other) const;
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \struct        TFStruct
+///  \brief         Struct to hold return value of count vectorizer,
+///                 consist of <dictid, # of appearances>
+///
+struct TFStruct {
+    std::uint32_t const DictionaryId;                              // dict id
+    std::uint32_t const Appearances;                               // number of appearances
+
+    TFStruct(std::uint32_t dictionaryid, std::uint32_t appearances);
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(TFStruct);
+    bool operator==(TFStruct const &other) const;
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \struct        TFIDFStruct
+///  \brief         Struct to hold return value of tfidf vectorizer,
+///                 consist of <dictid, tfidf value>
+///
+struct TFIDFStruct {
+    std::uint32_t const DictionaryId;                              // dict id
```

Comment:
>DictionaryId [](start = 24, length = 12)

Should not this have underscore as a member?

Reply:
This struct will be deprecated in next update